server/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -796,7 +796,7 @@ public void testReadCorruptedArraySize() throws IOException {
                     assertEquals(i, ints[i]);
                 }
                 EOFException eofException = expectThrows(EOFException.class, () -> streamInput.readIntArray());
-                assertEquals("tried to read: 100 bytes but only 40 remaining", eofException.getMessage());
+                assertEquals("tried to read: 100 bytes but this stream is limited to: 82", eofException.getMessage());
```

Comment:
Why did this change?

Reply:
It's a different stream now. Instead of `BytesReferenceStream` which generated the previous exception message it's backed by a `ByteArrayInputStream` now.